api/db-cdr/generate-cdr/libproject/genomic_manifests.rb

```diff
@@ -0,0 +1,159 @@
+require "csv"
+require "date"
+
+PROD_SOURCE_CONFIG = {
+  # Optional: to speed up iteration with this script, download and run against a local directory instead.
+  :wgs_aw4_prefix => "gs://prod-drc-broad/AW4_wgs_manifest/AoU_DRCB_SEQ_"
+}
+
+
+FILE_ENVIRONMENTS = {
+  "all-of-us-rw-preprod" => {
+    :source_config => PROD_SOURCE_CONFIG,
+  },
+  "all-of-us-rw-prod" => {
+    :source_config => PROD_SOURCE_CONFIG
+  }
+}
+
+FILE_PREFIX_REPLACEMENTS = {
+  :wgs_cram => {
+    # Match/replace the GC filename; this works for both .cram and .cram.crai
+    :matchSourceName => /^[^\/]+\.cram/,
+    :destNameFn => ->(rid) { "wgs_#{rid}.cram" }
+  }
+}
+
+def _aw4_filename_to_datetime(aw4_prefix, f)
+  common = Common.new
+
+  date_match = f.delete_prefix(aw4_prefix).match(/([\d-]+)(_\d+)?.csv/)
+  unless date_match
+    common.warning "AW4 filename does not match expected date format, assuming old: #{f}"
+    return DateTime.new(1970)
+  end
+  date_part = date_match[1]
+  begin
+    return DateTime.strptime(date_part, "%Y-%m-%d-%H-%M-%S")
+  rescue Date::Error
+    common.warning "Found date-part in AW4 filename but cannot parse it, assuming old: #{date_part}"
+    return DateTime.new(1970)
+  end
+end
+
+def _read_research_id_csvs(aw4_prefix, rids)
+  common = Common.new
+
+  ridset = rids.to_set
+
+  # See the CDR playbook for more documentation on AW4s:
+  # https://docs.google.com/document/d/1St6pG_EUFB9oRQUQaOSO7a9UPxPkQ5n4qAVyKF9j9tk/edit#
```

Comment:
this link did not enlighten me about AW4 - how about a brief summary comment at the top of the file?

Reply:
Added a longer comment to the top of the file, PTAL

Reply:
very nice, thank you